### PR TITLE
[WIP] Add ability to use existing data packages

### PR DESCRIPTION
**Pull request by ethanwhite**
Downloads Frictionless Data data packages from the web and adds
them to the retriever. Accomplished by:

* Retrieving the JSON metadata file
* Modifying it to work with the retriever
* Writing it to the scripts directory

New packages are added by adding a name-url pair to
scripts/datapackages.yml.

### CODE REVIEW

**Comment by ethanwhite** [comment]
The basics of this are working. Running:

```
from retriever.lib.get_datapackages import get_dps, download_dps_json
dps = get_dps()
download_dps_json(dps)
```

Will grab the json data package file, modify it to run with the retriever, and add it to the `~/.retriever/scripts` directory.

If other groups data package files looked exactly like ours we'd just need to hook this code up, but they don't, and I think this is going to mean a fair bit of additional work. E.g., there are lots of fields that are allowed in the json according to the spec that we don't process explicitly in `compile.py`. We currently pass those through unquoted, but many of them are strings and so this leads to errors loading the resulting `.py` scripts. This is the first thing to solve, but I suspect there will be a number of others.

File: retriever/lib/repository.py [hunk 1]
```diff
@@ -26,6 +27,10 @@ def download_from_repository(filepath, newpath, repo=REPOSITORY):
         raise
         pass
 
+def download_external_dps():
```
Comment (henrykironde) [inline retriever/lib/repository.py]: Some clean up. 
- [ ] more line space before the function.

File: retriever/lib/get_datapackages.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    official frictionless data spec. This converts frictionless data types to
+    retriever types.
+
+    """
+
+    types = {'string': 'char',
+             'number': 'double',
+             'integer': 'int',
+             'date': 'char'
+    }
```
Comment (henrykironde) [inline retriever/lib/get_datapackages.py]: - [ ]  Closing bracket not indented well
- [ ] Double spacing between functions
- [ ] Do we need the extra libs `urlparse, urlunparse, ParseResult`

**Comment by zhangcandrew** [comment]
Closing this pr now that changes have been incorporated into PR #1010. Feel free to open this PR up again if anything new comes to light!

**Pull request closed without merging**